IYY-283: Site editors can create a ½ - ½ layout | IYY-284: Site editors can create a ⅓ - ⅓ - ⅓ layout

joetower opened the pull request:
## [IYY-283: Site editors can create a ½ - ½ layout](https://fourkitchens.clickup.com/t/36718269/IYY-283) | [IYY-284: Site editors can create a ⅓ - ⅓ - ⅓ layout](https://fourkitchens.clickup.com/t/36718269/IYY-284)

### Description of work
- Adds new layout component
- Adds layout option for `fifty-fifty`
- Adds layout option for `thirty-thirty-thirty`
- Adds layout option for `two-thirds-one-thirds`
- - Requires [Main Repo PR-826](https://github.com/yalesites-org/yalesites-project/pull/826) and [Atomic PR-291](https://github.com/yalesites-org/atomic/pull/291)

### Testing Link(s)
- [ ] Navigate to the [Layout Component Story](https://deploy-preview-442--dev-component-library-twig.netlify.app/?path=/story/organisms-layouts--layout)

### Functional Review Steps
- [x] Navigate to: https://deploy-preview-442--dev-component-library-twig.netlify.app/?path=/story/organisms-layouts--layout
- [x] Verify you can choose:
- [x] Layout - `fifty-fifty`
- [x] Layout - `thirty-thirty-thirty`
- [x] Layout - `two-thirds-one-thirds`
- [x] Background Them - `one`, `two`, `three`, `four`, 
- [x] Divider - `true` or `false`
- [x] Note that if using a `fifty-fifty` the second column has an accordion to illustrate that it can only be added on a fifty-fifth layout

[IYY-layouts.webm](https://github.com/user-attachments/assets/28db0690-e51f-472f-98b4-130f1b3e1d6d)


### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

miketullo95 reviewed (approved):
looks amazing! do you have the list of approved blocks for each section? I know in the past we had a table to keep track of them

joetower commented:
@miketullo95 Yes, the approved list of components is as follows: 

### Fifty-Fifty
- Text
- Accordion
- Media
- Link Grid (adding in a separate PR - in-progress now)

### Thirty-Thirty-Thirty
- Text
- Media

miketullo95 commented:
Potential easy wins (not related to IYY project, just brainstorming here)

### 50/50
- Button
- Quote
- In-line message
- Tiles ?

### 70/30
- Views (70)
- Tiles (70)
- In-line message (70, maybe 30?)
- Image Grid (70)
- Image Gallery (70)
- Image (30)

joetower commented:
@miketullo95 Could you capture that request in a ticket? 

It would require a decent amount of work and definitely wouldn't be quick :laughing: .

miketullo95 commented:
@joetower oh, this wasn't specifically a request for this PR/work package, just thinking out-loud and wanted a record of potential stuff that can be taken on in the future 😆. Will definitely make a ticket for it now

joetower commented:
@miketullo95 Okay, sounds great! Thank you.

miketullo95 commented:
@joetower feedback from the university printer:

- [ ] it looks like the divider for the 70/30 is slightly narrower than the others? could all of them be that thinner style?
- [ ] change the opacity of the dividers to put less emphasis on them, try out 70%, then 50%

joetower commented:
@miketullo95 Apologies for sending you to another PR, but could you please test your changes here: https://github.com/yalesites-org/component-library-twig/pull/445

I made newer changes to the layout files in that branch and wanted to keep subsequent updates in the same place. 

The divider was a bit thinner in the `seventy-thirty` layout because the CSS `gap` property seems to affect the overall calculated width (of the divider since it uses rems). It's odd. I added a bit extra space to the divider in that context and it appears to have helped.

Also, I added `50%` opacity to the dividers. :partying_face:

miketullo95 commented:
@joetower amazing as always! and apologies, forgot that those requests were CL based 😆

codechefmarc reviewed (approved):
I've been working on this PR for the back-end, wiring, and UI parts to this ticket and all of this looks great to me. Approved!
